app/master/cluster_master.py

```diff
@@ -35,6 +38,10 @@ def __init__(self):
         self._build_request_handler.start()
         self._slave_allocator = SlaveAllocator(self._scheduler_pool)
         self._slave_allocator.start()
+
+        self._heartbeat_frequency = Configuration['heartbeat_frequency'] * Configuration['heartbeat_count_threshold']
```

Comment:
Nit: `heartbeat_frequency` should be named `heartbeat_period` or `heartbeat_interval`. A frequency has units 'something per second', a period has units of seconds. I expect heartbeat_frequency to have units 'beats per second'. When I multiplied the frequency by a time interval, I would expect to get the number of beats that should occur in that time interval. That's not what this variable represents.